src/v/cloud_storage/partition_manifest.cc

```diff
@@ -510,7 +532,8 @@ bool partition_manifest::add(
         // to the manifest or if all data was removed previously.
         _start_offset = meta.base_offset;
     }
-    move_aligned_offset_range(meta.base_offset, meta.committed_offset);
+    auto newly_replaced_begin = move_aligned_offset_range(
```

Comment:
This could be a bit more readable if the method will return total size of replaced segments. The method has to iterate though them so it can compute this value without much overhead.

Reply:
Not sure I understand. Are you suggesting to make `move_aligned_offset_range` return the new size of `_replaced`, cache the old size in `add`, and then iterated between the two indices to update the size?

Reply:
I'm suggesting to return a total size of replaced segments. If the segment replaces segment A and B then this method will move them to `_replaced` list. While doing this it can easily accumulate sizes of replaced segment (the one being moved to `_replaced`). This will make the call site a bit more simple.